autorag/data/chunk/llama_index_chunk.py

```diff
@@ -14,42 +19,66 @@ def llama_index_chunk(
 	chunker: NodeParser,
 	file_name_language: Optional[str] = None,
 	metadata_list: Optional[List[Dict[str, str]]] = None,
-) -> Tuple[List[str], List[str], List[Dict[str, Any]]]:
+	batch: int = 8,
+) -> Tuple[
+	List[str], List[str], List[str], List[Tuple[int, int]], List[Dict[str, Any]]
+]:
 	"""
 	Chunk texts from the parsed result to use llama index chunk method
 
 	:param texts: The list of texts to chunk from the parsed result
 	:param chunker: A llama index NodeParser(Chunker) instance.
 	:param file_name_language: The language to use 'add_file_name' feature.
-		You need to set one of 'English' and 'Korean'
-		The 'add_file_name' feature is to add a file_name to chunked_contents.
-		This is used to prevent hallucination by retrieving contents from the wrong document.
-		Default form of 'English' is "file_name: {file_name}\n contents: {content}"
+	    You need to set one of 'English' and 'Korean'
+	    The 'add_file_name' feature is to add a file_name to chunked_contents.
+	    This is used to prevent hallucination by retrieving contents from the wrong document.
+	    Default form of 'English' is "file_name: {file_name}\n contents: {content}"
 	:param metadata_list: The list of dict of metadata from the parsed result
-	:return: tuple of lists containing the chunked doc_id, contents, and metadata
+	:param batch: The batch size for chunk texts. Default is 8
+	:return: tuple of lists containing the chunked doc_id, contents, path, start_idx, end_idx and metadata
 	"""
-	# set documents
-	documents = [
-		Document(text=text, metadata=metadata)
-		for text, metadata in zip(texts, metadata_list)
+	tasks = [
+		llama_index_chunk_pure(text, chunker, file_name_language, meta)
+		for text, meta in zip(texts, metadata_list)
 	]
+	loop = get_event_loop()
+	results = loop.run_until_complete(process_batch(tasks, batch))
+
+	doc_id, contents, path, start_end_idx, metadata = (
+		list(chain.from_iterable(item)) for item in zip(*results)
+	)
+
+	return list(doc_id), list(contents), list(path), list(start_end_idx), list(metadata)
 
-	# chunk documents
-	chunk_results = chunker.get_nodes_from_documents(documents=documents)
+
```

Comment:
Why did you use async at here? 
There are no await in the code, so async will be useless.
It only add overhead to the event loop